google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineFacetConfiguration.java

```diff
@@ -69,19 +71,11 @@ public void writeExternal(Element element) throws WriteExternalException {
    * returned and therefore not considered to be "managed".
    */
   public Set<AppEngineStandardMavenLibrary> getLibraries(@NotNull Project project) {
-    LibraryTable libraryTable = LibraryTablesRegistrar.getInstance().getLibraryTable(project);
-    Library[] libraries =  libraryTable.getLibraries();
-
-    Set<AppEngineStandardMavenLibrary> configuredLibraries = Sets.newHashSet();
-    for (Library configuredLibrary : libraries) {
-      AppEngineStandardMavenLibrary mavenLibrary
-          = AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(configuredLibrary.getName());
-      if (mavenLibrary != null) {
-        configuredLibraries.add(mavenLibrary);
-      }
-    }
-
-    return configuredLibraries;
+    return Arrays.stream(
+        LibraryTablesRegistrar.getInstance().getLibraryTable(project).getLibraries())
+        .map(library ->
+            AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(library.getName()).get())
```

Comment:
Yes. Turns out that the recommended way appears to be:

libraries.stream()
        .filter(JCheckBox::isSelected)
        .map(
            library ->
                AppEngineStandardMavenLibrary.getLibraryByDisplayName(library.getText()))
        .filter(Optional::isPresent)
        .map(Optional::get)
        .collect(toSet());

The Optional creator discusses it in http://stackoverflow.com/questions/22725537/using-java-8s-optional-with-streamflatmap. Java 9 will enable Optional.stream(), which makes flatMap here work as intended.